generate/resources/couchbase-edge-server/README.md

```diff
@@ -1,16 +1,101 @@
+```
+
+At this point you should be able to send a HTTP request to the Edge Server on port `59840` using curl:
 
-1. Start Edge Server container with the default configuration file
 ```
-$ docker run --name edge-server couchbase/edge-server
+$ curl http://localhost:59840
+{"couchdb":"Welcome","vendor":{"name":"CouchbaseEdgeServer","version":"0.0.0 ()"},"version":"CouchbaseEdgeServer/0.0.0 () CouchbaseLiteCore/1.0.0 (21)"}
 ```
 
-2. Start Edge Server container with an external configuration file
+This setup is minimal and launches with an empty database, though any changes will be persisted.  Some key container paths are noted in the following diagram:
+
+![Default Docker Layout](./diagrams/docker-nomount.png)
+
+Where the bin and lib directories contain the necessary executable data to run the server, `/etc/config.json` is the configuration file that was used at startup, and `/var/databases` stores the resulting database specified by the configuration.
+
+## Viewing Logs
+You can view the Sync Gateway logs via the `docker logs` command:
+
+```
+$ docker logs edge-server
+2025-01-24T01:59:00.276Z        Using server TLS certificate: CN=127.0.0.1
+2025-01-24T01:59:00.277Z        Starting Couchbase Edge Server 0.0.0 ()
+2025-01-24T01:59:00.339Z        Sharing database https://localhost:59840/db/ from file /opt/couchbase-edge-server/var/databases/example.cblite2
+etc ...
 ```
-$ docker run --name edge-server -d -v /tmp/my-edge-server.json:/tmp/my-edge-server.json couchbase/edge-server /tmp/my-edge-server.json
+
+# Customizing Edge Server configuration
+
+## Using a Docker volume
+
+The default configuration is useful for trying out the server, but a production deployment is going to be a little more involved in its creation.  A production deployment should contain persistent files that are outside of the container, as shown in the following diagram:
+
+![Production Docker](./diagrams/docker-mount.png)
+
+**Step - 1 :** Prepare a local working directory somewhere on your machine.  These steps will refer to your working directory as `$LOCALPATH`.  Inside this directory, create two folders called `etc` and `databases`.
+
+**Step - 2 :** Start the docker container in interactive mode with the previously created `etc` directory mounted.  This is accomplished by setting the entry point of the container to `bash`. 
+
 ```
+docker run -it --rm --entrypoint=bash -v $LOCALPATH/etc:/opt/couchbase-edge-server/etc couchbase/edge-server
+```
+
+**Step - 3 :** Create a TLS key and certificate for use with the server.  Normally, this would be done through your provider and the instructions are out of scope for this README, but for testing and tutorial purposes the edge server can create some self signed certificates for you.  The following command will create a cert and key file in the mounted `etc` directory (the default working directory of the container)
+
+```
+couchbase-edge-server --create-cert 127.0.0.1 cert.pem key.pem
```

Comment:
Using `127.0.0.1` will only work if the client is on the same machine. Probably you should call this `$HOSTNAME` or something, and tell the reader to substitute the hostname (or IP address) that clients will be using.

Also, you should warn the reader that `key.pem` contains a sensitive private key, and to ensure it's not readable by untrusted accounts.